rpm/opensuse.go

```diff
@@ -51,11 +58,71 @@ func (b *OpenSUSEBackend) GetKernelHeaders(directory string) error {
 	return b.dnfBackend.GetKernelHeaders(pkgNevra, directory)
 }
 
+// On newer versions of openSUSE, the repos are type yast2 instead of type yum.
+// For now, librepo only supports yum, so we need to convert the yast2 repo baseurls
+// to 'yum' format. Without this, attempting to download the repomd.xml file
+// results in a 404 error.
+func yumifyRepositories(reposDir string, logger types.Logger) (string, error) {
+	tmpDir, err := ioutil.TempDir("", "yum.repos.d")
+	if err != nil {
+		return "", err
+	}
+
+	repoFiles, err := filepath.Glob(reposDir + "/*.repo")
+	if err != nil {
+		os.RemoveAll(tmpDir)
+		return "", err
+	}
+
+	for _, repoFile := range repoFiles {
+		logger.Debugf("Reading repo file '%s'", repoFile)
+		cfg, err := ini.Load(repoFile)
+		if err != nil {
+			logger.Warnf("Fail to read file '%s': %v", repoFile, err)
```

Comment:
Do you need a `continue` here?

Reply:
We don't want to `continue` because we still want to try to use `SaveTo` to write the contents of this file to the new temp directory if there was an error. An alternative strategy could be to copy the contents of the repos directory to the temp directory, then perform this loop, in that case we would want to `continue` as soon as we have issues parsing the file. I'm fine with either approach

Reply:
If there is an error, then `cfg` will be `nil`.

Reply:
If your intent is to copy the file over unmodified in the face of an error, I think you need some additional code.

Reply:
Oops, you're right. I'll change this